fix: Ensure nil registry credentials config when no values are set

byDimasik opened the pull request:
Resolves https://github.com/k0rdent/2a-internal/issues/75

zerospiel commented on internal/sveltos/profile_test.go:
```diff
@@ -44,3 +46,58 @@ func Test_priorityToTier(t *testing.T) {
 	}
 }
+
+func Test_emptyRegistryCredentialsConfig(t *testing.T) {
```
both test functions could be flattened into a single

byDimasik replied:
I would like to keep them separate for easier maintenance. There are scenarios when we expect this config to be nil, and all such scenarios should be covered in a separate test for better visibility. The whole bug was about the config not being nil when it's supposed to be, so, having separate tests for nil and non-nil scenarios ensures we can easily cover all nil cases in one place.

zerospiel reviewed (commented):
lgtm but someone else from the ksm should also take a look

a13x5 reviewed (commented):
LGTM, but please rebase and resolve all comments before merging

byDimasik commented:
@zerospiel @BROngineer @a13x5 

Thank you for the reviews! I've addressed all the comments and rebased main to resolve conflicts. Could you please take another look and put a ✅ if you have no more questions so I can proceed with merging? Thanks again!
